### PR TITLE
SHO-81: Menu page

### PR DESCRIPTION
**Pull request by saulprl**
## Describe your changes

This PR adds a `/menu` page which shows all of the products in the sales channel grouped by collection.

## Task URL (Asana, Jira, etc...)

[SHO-81](https://cultoperrocafe.nifty.pm/l/nhK8tAtLKpAo)

## What was done in this pull request

- [x] Added `<MenuPage />` component (I believe we should look into renaming the `<Menu />` component to something like `<MenuSection />` in order to let this one be just `<Menu />`.
- [x] Added `/menu` route with its own GraphQL query.
- [x] Storybook for `<MenuPage />` with dummy data.
- [ ] Still unable to type `_Link` or `NavBarLink`s.

## Demo / Screenshot / Video

[SHO-81-1.webm](https://user-images.githubusercontent.com/39208827/236588051-c6518afe-1d9c-429a-82b7-0cdff37bb09b.webm)

### CODE REVIEW

File: app/components/templates/menu-page/menu-page.component.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  const {
+    minVariantPrice: {amount: minPrice, currencyCode: minCurrency},
+    maxVariantPrice: {amount: maxPrice, currencyCode: maxCurrency},
+  } = priceRange;
+
+  const isActualRange = minPrice !== maxPrice;
+
+  const price = isActualRange
+    ? `${minPrice}0 ${minCurrency} - ${maxPrice}0 ${maxCurrency}`
+    : `${maxPrice}0 ${maxCurrency}`;
```
Comment (arielplas) [inline app/components/templates/menu-page/menu-page.component.tsx]: Nice, but please use Javascript's built in currency handler (Number format). take a look a [this doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat)

File: app/components/templates/menu-page/menu-page.component.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  ProductsGrid,
+} from './menu-page.styles';
+import {Divider} from '../../atoms/divider';
+import {Hr} from '../../atoms/hr';
+import {Paragraph} from '../../atoms/paragraph';
+import {Subtitle} from '../../molecules/subtitle';
+
+export interface MenuProductPriceRange {
+  minVariantPrice: {amount: string; currencyCode: string};
+  maxVariantPrice: {amount: string; currencyCode: string};
```
Comment (arielplas) [inline app/components/templates/menu-page/menu-page.component.tsx:18]: Why is the amount a string? is this what Shopify sends it as?

Comment (saulprl) [reply app/components/templates/menu-page/menu-page.component.tsx:18]: I'm pretty sure it is but I'll check again just in case.

Comment (saulprl) [reply app/components/templates/menu-page/menu-page.component.tsx:18]: Shopify handles the amount as a Decimal but in the editor it gives me an error whenever I define it as a number.

Looked it up and `amount` is serialized as a string: https://shopify.dev/docs/api/storefront/2023-01/scalars/Decimal

File: app/components/templates/menu-page/menu-page.component.tsx (hunk 1 above)
Comment (saulprl) [reply app/components/templates/menu-page/menu-page.component.tsx]: Didn't know about this one, thanks!

File: app/components/templates/menu-page/menu-page.component.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    <Divider />
+  </MenuPageContainer>
+);
+
+const ProductItem: FC<ProductProps> = ({title, src, priceRange}) => {
+  const formatPrice = (price: string, currency: string) =>
+    `${Intl.NumberFormat('es-MX', {
+      style: 'currency',
+      currency,
+    }).format(Number(price))} ${currency}`;
```
Comment (arielplas) [inline app/components/templates/menu-page/menu-page.component.tsx]: Extract this NumberFormat into a separate file. No need to re-create it on every render.

maybe into a folder/file like -> `utils/formatters.ts`.

```typescript
const currencyFormatter = Intl.NumberFormat('es-MX', {
  style: 'currency',
  currency,
})
```

File: app/components/templates/menu-page/menu-page.component.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  const {
+    minVariantPrice: {amount: minPrice, currencyCode: minCurrency},
+    maxVariantPrice: {amount: maxPrice, currencyCode: maxCurrency},
+  } = priceRange;
+
+  const isActualRange = minPrice !== maxPrice;
+
+  const price = isActualRange
+    ? `${formatPrice(minPrice, minCurrency)} - ${formatPrice(
```
Comment (arielplas) [inline app/components/templates/menu-page/menu-page.component.tsx]: No need to create a separa function, just use 

```typescript
 currencyFormatter.format(...)
```

**Review by imariel2d — DISMISSED** [review]
Looks fine, just extract the formatter and merge it :)

**Review by imariel2d — APPROVED** [review]
LGTM :)

**Pull request merged**